adapter/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,276 @@
+	r.RegisterAttributesGeneratorBuilder(newBuilder(newCacheFromConfig))
+}
+
+func newBuilder(cacheFactory controllerFactoryFn) *builder {
+	stopChan := make(chan struct{})
+	return &builder{adapter.NewDefaultBuilder(name, desc, conf), stopChan, nil, sync.Once{}, cacheFactory}
+}
+
+func (b *builder) Close() error {
+	close(b.stopChan)
+	return nil
+}
+
+func (*builder) ValidateConfig(c adapter.Config) (ce *adapter.ConfigErrors) {
+	params := c.(*config.Params)
+	if len(params.SourceUidInputName) == 0 {
+		ce = ce.Appendf("source_uid_input_name", "field must be populated")
+	}
+	if len(params.TargetUidInputName) == 0 {
+		ce = ce.Appendf("target_uid_input_name", "field must be populated")
+	}
+	if len(params.OriginUidInputName) == 0 {
+		ce = ce.Appendf("origin_uid_input_name", "field must be populated")
+	}
+	if len(params.SourcePrefix) == 0 {
+		ce = ce.Appendf("source_prefix", "field must be populated")
+	}
+	if len(params.TargetPrefix) == 0 {
+		ce = ce.Appendf("target_prefix", "field must be populated")
+	}
+	if len(params.OriginPrefix) == 0 {
+		ce = ce.Appendf("origin_prefix", "field must be populated")
+	}
+	if len(params.LabelsValueName) == 0 {
+		ce = ce.Appendf("labels_value_name", "field must be populated")
+	}
+	if len(params.PodIpValueName) == 0 {
+		ce = ce.Appendf("pod_ip_value_name", "field must be populated")
+	}
+	if len(params.PodNameValueName) == 0 {
+		ce = ce.Appendf("pod_name_value_name", "field must be populated")
+	}
+	if len(params.HostIpValueName) == 0 {
+		ce = ce.Appendf("host_ip_value_name", "field must be populated")
+	}
+	if len(params.NamespaceValueName) == 0 {
+		ce = ce.Appendf("namespace_value_name", "field must be populated")
+	}
+	if len(params.ServiceAccountValueName) == 0 {
+		ce = ce.Appendf("service_account_value_name", "field must be populated")
+	}
+	if len(params.ServiceValueName) == 0 {
+		ce = ce.Appendf("service_value_name", "field must be populated")
+	}
+	if len(params.PodLabelForService) == 0 {
+		ce = ce.Appendf("pod_label_name", "field must be populated")
+	}
+	return
+}
+
+func (b *builder) BuildAttributesGenerator(env adapter.Env, c adapter.Config) (adapter.AttributesGenerator, error) {
+	var clientErr error
+	paramsProto := c.(*config.Params)
+	b.once.Do(func() {
+		refresh := paramsProto.CacheRefreshDuration
+		controller, err := b.newCacheControllerFn(paramsProto.KubeconfigPath, refresh, env.Logger())
+		if err != nil {
+			clientErr = err
+			return
+		}
+		b.pods = controller
+		go b.pods.Run(b.stopChan)
+		// ensure that any request is only handled after
+		// a sync has occurred
+		cache.WaitForCacheSync(b.stopChan, b.pods.HasSynced)
+	})
+	if clientErr != nil {
+		return nil, clientErr
+	}
+	kg := &kubegen{
+		log:    env.Logger(),
+		pods:   b.pods,
+		params: *paramsProto,
+	}
+	return kg, nil
+}
+
+func newCacheFromConfig(kubeconfigPath string, refreshDuration time.Duration, log adapter.Logger) (cacheController, error) {
+	log.Infof("getting kubeconfig from: %#v", kubeconfigPath)
```

Comment:
does this work while running in-cluster? Does it has the expected behaviour when you leave the kubeconfigPath blank ?